javascript/javascript1/week2/main.js

```diff
@@ -0,0 +1,85 @@
+function getFullname(firstname, surname, formal, male) {
+  let full = "";
+  if (firstname == undefined || surname == undefined) {
+    return "Please Enter your Fullname";
+  } else if (formal) {
+    if (male) {
+      full = "Lord " + firstname + " " + surname;
+    } else {
+      full = "Lady " + firstname + " " + surname;
+    }
+  } else {
+    full = firstname + " " + surname;
+  }
+  return full;
+}
+
+function getEventWeekday(todaysWeekday, days) {
+  let weeksDays = [
+    "monday",
+    "tuesday",
+    "wednesday",
+    "thursday",
+    "friday",
+    "saturday",
+    "sunday",
+  ];
+  for (let i = 0; i < 7; i++) {
+    if (todaysWeekday == weeksDays[i]) {
+      let p = (days % 7) + i;
+      return weeksDays[p % 7];
+    }
+  }
```

Comment:
I think you are doing something rather clever by taking the input of a weekday string and converting it. You are also using only lowercase weekdays in your list, which is helpful if you also lowercase the input :) I put in "Thursday" and your code could not recognise it. Otherwise, good solution. You might want to look into key/value pairs, or Dictionaries like here https://pietschsoft.com/post/2015/09/05/javascript-basics-how-to-create-a-dictionary-with-keyvalue-pairs This would allow you to create a dictionary where `weekday["monday"]` would return `1` This is beyond the scope of the exercise, but I thnk you'd find it fun :)